src/db_state.rs

```diff
@@ -156,6 +156,7 @@ pub(crate) struct COWDbState {
     pub(crate) imm_memtable: VecDeque<Arc<ImmutableMemtable>>,
     pub(crate) imm_wal: VecDeque<Arc<ImmutableWal>>,
     pub(crate) core: CoreDbState,
+    pub(crate) error: Option<SlateDBError>,
```

Comment:
wdyt about putting this in `DbInner` rather than `DbState`. I had intended `DbState` to be used for orchestrating changes to the data stored in the database. Some practical reasons why this might be problematic:
- I expect it's going to become a bit more complicated when we add checkpoints because the data in here will need to be associated with a checkpoint and we'll have to maintain that relationship as well. Error tracking is orthogonal to this.
- Range scans will work by acquiring a snapshot and then processing the snapshot. The error here is tied to this snapshot and so the range scan wouldn't observe any errors that were hit after the range scan started. Which seems bad for long range scans.

Reply:
Let me change that to see how it looks like.

Reply:
I've updated the code to move the error to `DbInner`, it looks fine too.